node/src/actors/chain_manager/mod.rs

```diff
@@ -403,6 +414,31 @@ impl ChainManager {
                     hash_prev_block: block_hash,
                 };
 
+                // Get VRF context
+                let vrf_ctx = match self.vrf_ctx.as_mut() {
+                    Some(x) => x,
+                    None => {
+                        error!("No VRF context available");
+                        return;
+                    }
+                };
+
+                // If synching from genesis, start consolidating as genesis hash.
+                // Else, take hash the proof to obtain the new vrf output.
+                let vrf_input = match block_epoch {
+                    0 => CheckpointBeacon {
+                        checkpoint: block_epoch,
+                        hash_prev_block: block_hash,
+                    },
+                    _ => {
+                        let proof_hash = block.block_header.proof.proof_to_hash(vrf_ctx).unwrap();
+                        CheckpointBeacon {
+                            checkpoint: block_epoch,
+                            hash_prev_block: proof_hash,
+                        }
+                    }
```

Comment:
I have my reservations about usage of `CheckpointBeacon` for purposes other than acting as such.

I mean, it sounds counter-intuitive to me that `hash_prev_block` contains something that is not literally the hash of the previous block.

EDIT: this is a code only concern under the assumption that the `Hasher` that we are using doesn't take into account the name of the fields but only the types and order in which they were defined.

Reply:
Understood!